src/test/java/picard/sam/ReorderSamTest.java

```diff
@@ -0,0 +1,194 @@
+package picard.sam;
+
+import htsjdk.samtools.*;
+import htsjdk.variant.utils.SAMSequenceDictionaryExtractor;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import picard.cmdline.CommandLineProgramTest;
+import picard.util.SequenceDictionaryUtils;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Created by farjoun on 1/12/18.
+ */
+public class ReorderSamTest extends CommandLineProgramTest {
+
+    @Override
+    public String getCommandLineProgramName() {
+        return ReorderSam.class.getSimpleName();
+    }
+
+    @DataProvider(name = "testDictionaryData")
+    public Iterator<Object[]> testDictionaryData() throws IOException {
```

Comment:
I'm pretty sure this gets called at least once per test that requires the data. If the data is constant across all tests and is not modified by the tests it would be more efficient to create it in a `@BeforeClass` instead, since this does file I/O and other things that may take a while.

Reply:
no more file IO in the data provider...so OK so leave as is?